src/com/google/javascript/jscomp/DependencyOptions.java

```diff
@@ -151,7 +152,88 @@ boolean shouldDropMoochers() {
     return pruneDependencies && dropMoochers;
   }
 
-  Collection<String> getEntryPoints() {
+  Collection<ModuleIdentifier> getEntryPoints() {
     return entryPoints;
   }
+
+  /**
+   * Basic information on an entry point module.
+   * While closure entry points are namespaces,
+   * ES6 and CommonJS entry points are file paths
+   * which are normalized to a closure namespace.
+   *
+   * This class allows error messages to the user to
+   * be based on the input name rather than the
+   * normalized version.
+   */
+  public static final class ModuleIdentifier {
+    private final String name;
+    private final String closureNamespace;
+    private final String moduleName;
+
+    /**
+     * @param name as provided by the user
+     * @param closureNamespace entry point normalized to a closure namespace
+     * @param moduleName For closure namespaces, the module name may be different than
+     *     the namespace
+     */
+    ModuleIdentifier(String name, String closureNamespace, String moduleName) {
+      this.name = name;
+      this.closureNamespace = closureNamespace;
+      this.moduleName = moduleName;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public String getClosureNamespace() {
+      return closureNamespace;
+    }
+
+    public String getModuleName() {
+      return moduleName;
+    }
+
+    public String toString() {
+      if (closureNamespace.equals(moduleName)) {
+        return closureNamespace;
+      }
+      return moduleName + ":" + closureNamespace;
+    }
+
+    /**
+     * @param name Closure namespace used as an entry point. May start
+     *     "goog:" when provided as a flag from the command line.
+     *
+     * Closure entry points may also be formatted as:
+     *     'moduleName:name.space'
```

Comment:
This is after the "goog:" right? I think the comment would be clearer if it said 'goog:moduleName:name.space'